packages/toplevel-alias/contracts/TopLevelAliasRegistry.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title TopLevelAliasRegistry
+ * @dev A contract for managing a registry of aliases mapped to names.
+ * This contract allows the owner to set aliases for given names, ensuring aliases end with '.eth',
+ * have at least three characters before '.eth', do not exceed a maximum length, and are not empty.
+ */
+contract TopLevelAliasRegistry is Ownable {
+    // Public mapping of names to aliases
+    mapping(string => string) public aliases;
```

Comment:
I think it could make sense to use the chainId as a key here. That makes it easier to query the contract and reduces storage costs.

Reply:
https://docs.ens.domains/ens-improvement-proposals/ensip-9-multichain-address-resolution